docs/library/spi_engine/index.rst

```diff
@@ -15,6 +15,7 @@ SPI Engine
    SPI Bus Interface<spi-bus-interface>
    Instruction Set Specification<instruction-format>
    Tutorial - PulSAR ADC<tutorial>
+   SPI Engine Execution Delays<spi_engine_delays>
```

Comment:
Because we are also detailing delays for other submodules

Reply:
Then it should be "Pipeline Delays", "Delays", "SPI Engine Delays", or "SPI Engine Framework Delays".
But not "Propagation Delays", because I think this term should be reserved to the time taken to a signal to propagate in a single clock cycle (e.g. between registers).

Reply:
I agree that "Propagation Delays" is definetivley not what this PR is. But I don't think it's written anywhere on the added doc is it?

Reply:
The title of the page however does say "SPI Engine Execution Delays", I think this might cause confusion. Changing.